backend/validation/electionSchema.js

```diff
@@ -1,8 +1,14 @@
 import { z } from "zod";
 
 export const electionSchema = z.object({
+  creatorId: z.string({ required_error: "Creator ID is required" }),
```

Comment:
creatorId should not be taken from the frontend. This would allow anyone to vote for any user (For instance with BurpSuite interceptor) which definitely shouldn't be allowed. Instead, the creatorId (userId) should be taken from the req.user object, which should be added to req, inside the isAuthenticated middleware if the authentication is successful. That's a common pattern in express. It would look something like this
```js
export const isAuthenticated = async (req, res, next) => {
    //...
    const user = jwt.verify(token, JWT_SECRET);
    if (!user) {
      return next(HttpError(401));
    }
    req.user = user;
    next();
    //...
};
```